### DISCUSSION TITLE
C# Omnisharp Broke w/ Recent Update

### DISCUSSION THREAD

**Discussion by mbo17210** [microsoft/vscode-discussions#811] [Extension Development QnA] [answered]
The recent update for VS Code seems to have broken Omnisharp for both me and someone else I know and we are not longer getting error, warning, syntax, etc underlining on our code. Everything else seems to be working fine - the code is still being colored properly, you can ctrl+click to navigate fast, the run unit test buttons are there, etc. I also don't know if it is tied to the issue, but the line `  "omnisharp.enableImportCompletion": true,` was auto-removed from my settings.json (adding it back doesn't seem to do anything. I've also tried uninstalling & reinstalling the C# extension without any luck. I was wondering if anyone else has experienced something similar or knows a potential solution.

**Comment by gjsjohnmurray** [comment]
I recommend checking the extension repository for related issues, then opening one there if you don't find an existing one.

This discussion forum is intended for extension developers rather than extension users.

**Comment by alefragnani** [comment, marked as answer]
Hi @mbo17210 ,

I'm not sure if you are aware of but, there was a big update in the C# extension (https://devblogs.microsoft.com/visualstudio/announcing-csharp-dev-kit-for-visual-studio-code/), which, beyond among other things, moved from Omnisharp a new Language Server (by default) under the hood, and could affect some scenarios.

If you didn't know about it, I suggest you to try out one of these approaches
* Enable Omnisharp, as detailed here https://github.com/dotnet/vscode-csharp#how-to-use-omnisharp, and retry
* Install an older version of the extension, prior to this update (I suggest v1.26.0), and retry
![image](https://github.com/microsoft/vscode-discussions/assets/3781424/79eb20b8-3cb7-4d4c-bd3a-108f0dd21d0a)


If you knew about it, then you should open an issue in the C# extension repo (https://github.com/dotnet/vscode-csharp/issues/new/choose)  providing as much detail as possible .

As commented by @gjsjohnmurray, this Discussions repo is focused for extension development for now.

Hope this helps

**Comment by Edmon02** [comment]
It's quite common for updates in development tools to occasionally introduce regressions. Given that multiple users are reporting issues with Omnisharp after the recent VS Code update, it's likely a compatibility problem with the latest VS Code API changes or the underlying.NET runtime environment that Omnisharp relies on. Besides downgrading, another immediate step could be to check the 'Output' panel in VS Code (View -> Output) and select 'Omnisharp' from the dropdown. This might provide more detailed error messages or logs that could help diagnose the specific issue. Sharing these logs in the Omnisharp extension's issue tracker on GitHub would be very helpful for the maintainers.
